vsphere_plugin_common/utils.py

```diff
@@ -78,12 +78,16 @@ def wrapper(**kwargs):
 
         # Support both node instance and relationship node instance CTX.
         ctx_node = _get_node(ctx)
+        ctx_instance = _get_instance(ctx)
 
         for key in requested_inputs:
-            if key in kwargs:
+            if key in ctx_instance.runtime_properties:
+                processed_kwargs[key] = ctx_instance.properties.get(key)
```

Comment:
shouldn't this be `ctx_instance.runtime_properties.get(key)`

Reply:
The reason I did it like this is to keep it in the style of the original.

Reply:
And also to accentuate that this is a condition - first runtime properties, then kwargs, then ctx node properties. Really, it shouldn't be like that. It should be kwargs, runtime properties, node properties, but because the kwargs default to node properties, I put them after runtime properties.